### PR TITLE
Describe EndDevice fields stored by IS/NS/AS/JS

### LINKED ISSUE #206: Document device registry APIs for JS, NS and AS
**Summary:** 
I am trying to implement the following CLI call:

dist/linux_amd64/ttn-lw-cli devices create caf dev3 --defaults=true \
--supports-join=true \
--join-eui=70B3D59520000000 --dev-eui=A81758FFFE031BF9 \
--root-keys.app-key.key=6E2DD93BA3DF1BE62D57C33D756594E4 \
--session.dev-addr=FC006A0C \
--session.keys.app-s-key.key=2412B7B219FCEE78BDFACB5944764116 \
--session.keys.f-nwk-s-int-key.key=57A976DF984AF3EC514B21E56C081510 \
--lorawan-version=1.0.2 --lorawan-phy-version=1.0.2-b \
--frequency-plan-id EU_863_870 \
--session.last-f-cnt-up=25860 \
--session.last-n-f-cnt-down=901

From what I understand, I need at least 3 http post:
one to the is:
/api/v3/applications/xxx/devices

one to the ns:
/api/v3/ns/applications/xxx/devices

one to the js:
/api/v3/js/applications/xxx/devices

and maybe one to the as ?
one to the ns:
/api/v3/as/applications/xxx/devices

Each call requires a subset of the info above and appropriate field_mask.paths. Correct ?

**Why do we need this?**
Simplify the API call or at least a clear documentation.

**What is already there? What do you see now?**
haven't seen much.

**What is missing? What do you want to see?**
a simple / single API to create / update devices.
If not possible because of the separation of components, at least some documentation about it.

**How do you propose to implement this?**
Ideally a single / simple call : http post or put.

**Environment:**
API call

**What can you do yourself and what do you need help with?**
I am not confortable enough with the components to write a go api, but I can try to write the doc when I sorted this out / or somebody has clarified the proper way to use the API.

### PR DESCRIPTION
**Pull request by htdvisser**
#### Summary

This PR extends the EndDevice API reference with an explanation about how end devices are stored in multiple registries and adds details on (some of) the fields stored by each registry.

Closes #206

#### Notes for Reviewers

This is not intended to be complete, just enough to get started with.

Feel free to add commits to this PR for anything essential that is still missing.

### CODE REVIEW

File: doc/content/reference/api/end_device.md [hunk 1]
```diff
@@ -4,59 +4,114 @@ description: ""
 weight: 7
 ---
 
+End devices are registered in multiple registries. The Identity Server has a registry with end device metadata, the Network Server's registry contains the MAC configuration and MAC state (including network session keys), the Application Server keeps payload formatters and application session keys, the Join Server keeps the root keys.
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]: Network session keys are not part of MAC state, so I think it's better to do the same as with AS:
```suggestion
End devices are registered in multiple registries. The Identity Server has a registry with end device metadata, the Network Server's registry contains the MAC configuration, MAC state, and network session keys, the Application Server keeps payload formatters and application session keys, the Join Server keeps the root keys.
```

File: doc/content/reference/api/end_device.md [hunk 2]
```diff
@@ -4,59 +4,114 @@ description: ""
...
+- `JsEndDeviceRegistry.Set` (Join Server, only for OTAA devices)
+- `NsEndDeviceRegistry.Set` (Network Server)
+- `AsEndDeviceRegistry.Set` (Application Server)
+
+When deleting end devices, we recommend deleting them in the reverse order
+
 ## The `EndDeviceRegistry` service
 
-{{< proto/method service="EndDeviceRegistry" method="Create" >}}
+The Identity Server's `EndDeviceRegistry` is the first place to register an end device. This registry stores the following [EndDevice fields](#message:EndDevice):
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]:
```suggestion
The Identity Server's `EndDeviceRegistry` is the first service, where end device is registered. This registry stores the following [EndDevice fields](#message:EndDevice):
```

File: doc/content/reference/api/end_device.md [hunk 3]
```diff
@@ -4,59 +4,114 @@ description: ""
...
-> TODO: [Document what fields to get from the JS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="JsEndDeviceRegistry" method="Delete" >}}
 
 ## The `NsEndDeviceRegistry` service
 
-{{< proto/method service="NsEndDeviceRegistry" method="Set" >}}
+The Network Server's `NsEndDeviceRegistry` stores the following [EndDevice fields](#message:EndDevice):
+
+- `ids` (with subfields)
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]: Is this list meant to be exhaustive or is it more of an example? If it's the former - this list is not complete, e.g. `last_dev_status_received_at` (and more)

File: doc/content/reference/api/end_device.md [hunk 4]
```diff
@@ -4,59 +4,114 @@ description: ""
...
-> TODO: [Document what fields to set to the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="EndDeviceRegistry" method="Delete" >}}
 
 ## The `JsEndDeviceRegistry` service
 
-{{< proto/method service="JsEndDeviceRegistry" method="Set" >}}
+OTAA devices are registered in the Join Server's `JsEndDeviceRegistry`. This registry stores the following [EndDevice fields](#message:EndDevice):
+
+- `ids` (with subfields)
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]: IDEM

File: doc/content/reference/api/end_device.md [hunk 5]
```diff
@@ -4,59 +4,114 @@ description: ""
...
 {{< proto/method service="NsEndDeviceRegistry" method="Get" >}}
 
-> TODO: [Document what fields to get from the NS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="NsEndDeviceRegistry" method="Delete" >}}
 
 ## The `AsEndDeviceRegistry` service
 
-{{< proto/method service="AsEndDeviceRegistry" method="Set" >}}
+The Network Server's `NsEndDeviceRegistry` stores the following [EndDevice fields](#message:EndDevice):
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]:
```suggestion
The Application Server's `AsEndDeviceRegistry` stores the following [EndDevice fields](#message:EndDevice):
```

File: doc/content/reference/api/end_device.md [hunk 6]
```diff
@@ -4,59 +4,114 @@ description: ""
...
 {{< proto/method service="EndDeviceRegistry" method="Update" >}}
 
-> TODO: [Document what fields to set to the IS.](https://github.com/TheThingsNetwork/lorawan-stack/issues/206)
-
 {{< proto/method service="EndDeviceRegistry" method="Delete" >}}
 
 ## The `JsEndDeviceRegistry` service
 
-{{< proto/method service="JsEndDeviceRegistry" method="Set" >}}
+OTAA devices are registered in the Join Server's `JsEndDeviceRegistry`. This registry stores the following [EndDevice fields](#message:EndDevice):
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]: The OTAA part is (and should be) already described above, maybe better to just be consistent with other components?
```suggestion
The Join Server's `JsEndDeviceRegistry` stores the following [EndDevice fields](#message:EndDevice):
```

Comment (htdvisser) [reply doc/content/reference/api/end_device.md]: I don't want to assume that all readers read all information at the top before they skip to here.

File: doc/content/reference/api/end_device.md (hunk 3 above)
Comment (htdvisser) [reply doc/content/reference/api/end_device.md]: Clarified with comment below

File: doc/content/reference/api/end_device.md [hunk 7]
```diff
@@ -4,59 +4,120 @@ description: ""
 weight: 7
 ---
 
+End devices are registered in multiple registries. The Identity Server has a registry with end device metadata, the Network Server's registry contains the MAC configuration, MAC state, and network session keys, the Application Server keeps payload formatters and application session keys, the Join Server keeps the root keys.
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]:
```suggestion
End devices are registered in multiple registries. The Identity Server has a registry with end device metadata, the Network Server's registry contains the MAC configuration, MAC state and network session keys, the Application Server keeps payload formatters and application session keys, the Join Server keeps the root keys.
```

File: doc/content/reference/api/end_device.md [hunk 8]
```diff
@@ -4,59 +4,120 @@ description: ""
...
+  - `nwk_key`
+- `net_id`
+- `network_server_address`
+- `network_server_kek_label`
+- `application_server_address`
+- `application_server_id`
+- `application_server_kek_label`
+- `claim_authentication_code` (when using [end device claiming]({{< relref "end_device_claiming.md" >}}))
+
+See the [EndDevice message](#message:EndDevice) and its sub-messages for additional fields that can be read from the Application Server's `AsEndDeviceRegistry`.
```
Comment (rvolosatovs) [inline doc/content/reference/api/end_device.md]:
```suggestion
See the [EndDevice message](#message:EndDevice) and its sub-messages for additional fields that can be read from the Join Server's `JsEndDeviceRegistry`.
```

Comment (htdvisser) [reply doc/content/reference/api/end_device.md]: Copy pasta great success in this PR 🤦‍♂

File: doc/content/reference/api/end_device.md (hunk 7 above)
Comment (htdvisser) [reply doc/content/reference/api/end_device.md]: Lol, I just committed your change suggestion 😝

**Pull request merged**